Library/Homebrew/unpack_strategy/cargo_crate.rb

```diff
@@ -0,0 +1,26 @@
+# typed: true
+# frozen_string_literal: true
+
+require_relative "tar"
+
+module UnpackStrategy
+  # Strategy for unpacking Cargo crates.
+  class CargoCrate < Tar
+    extend T::Sig
+
+    using Magic
+
+    sig { returns(T::Array[String]) }
+    def self.extensions
+      [".crate"]
+    end
+
+    def self.can_extract?(path)
+      return false unless Tar.can_extract?(path)
+
+      path.magic_number.match?(/\A\037\213\010\010/n) &&
+        path.magic_number.match?(/\.crate\b/n) &&
+        path.tar_files.grep(%r{\A[^/]+/Cargo.toml\Z}).any?
```

Comment:
Is this overkill?

Reply:
I think this check would only be needed if we *didn't* want to extract it but keep it as a `.crate` archive.

Reply:
Of course, if we remove this, it should go after `Tar` in the list.

Reply:
Just to make sure we're talking about the same thing, should we delete lines 21 through 23 or just line 23?

Reply:
Just 23 for me but I'll defer to @reitermarkus